src/dbnode/client/write_state.go

```diff
@@ -33,6 +33,18 @@ import (
 	"github.com/m3db/m3/src/x/serialize"
 )
 
+type shardStatus int64
```

Comment:
`shardStatus` is a bad name for this. it's being used to model which write acks count towards consistency - rename to something that reflects that.